pymc3/distributions/multivariate.py

```diff
@@ -1002,20 +1002,55 @@ def __init__(self, eta=None, n=None, p=None, transform='interval', *args, **kwar
             raise ValueError('Invalid parameter: please use eta as the shape parameter and '
                              'n as the dimension parameter.')
 
-        n_elem = int(n * (n - 1) / 2)
-        self.mean = np.zeros(n_elem, dtype=theano.config.floatX)
+        shape = n * (n - 1) // 2
+        self.mean = floatX(np.zeros(shape))
 
         if transform == 'interval':
             transform = transforms.interval(-1, 1)
 
-        super(LKJCorr, self).__init__(shape=n_elem, transform=transform,
+        super(LKJCorr, self).__init__(shape=shape, transform=transform,
                                       *args, **kwargs)
         warnings.warn('Parameters in LKJCorr have been rename: shape parameter n -> eta '
                       'dimension parameter p -> n. Please double check your initialization.',
                       DeprecationWarning)
         self.tri_index = np.zeros([n, n], dtype='int32')
-        self.tri_index[np.triu_indices(n, k=1)] = np.arange(n_elem)
-        self.tri_index[np.triu_indices(n, k=1)[::-1]] = np.arange(n_elem)
+        self.tri_index[np.triu_indices(n, k=1)] = np.arange(shape)
+        self.tri_index[np.triu_indices(n, k=1)[::-1]] = np.arange(shape)
+
+    def _random(self, n, eta, size=None):
```

Comment:
Yep. I ignored the size as the _random method here can only generate 1 slide of the random matrix.